crates/fuel-core/src/schema/balance.rs

```diff
@@ -79,9 +79,11 @@ impl BalanceQuery {
         Ok(balance)
     }
 
-    // TODO: This API should be migrated to the indexer for better support and
-    //  discontinued within fuel-core.
-    #[graphql(complexity = "query_costs().balance_query")]
+    #[graphql(
+        complexity = "query_costs().balance_query + query_costs().storage_iterator \
+        + (query_costs().storage_read * first.unwrap_or_default() as usize) * child_complexity \
+        + (query_costs().storage_read * last.unwrap_or_default() as usize) * child_complexity"
+    )]
```

Comment:
Actually, we want to have something like `count * child_complexity + count * storage_read`, not `storage_read * child_complexity`

Reply:
Please see here: https://github.com/FuelLabs/fuel-core/pull/2490/commits/afc133b3200feb2f0750266a0beb86f464b85d0d

This commit contains changes that are needed to support ~11k balances as sent from the Rust SDK and bail when more is requested.